crates/sdk/src/masp/utils.rs

```diff
@@ -153,13 +245,426 @@ where
                     extracted_masp_txs,
                 ));
             }
-            fetch_iter.next();
         }
 
         Ok(())
     }
 }
 
+/// MASP client implementation that queries data from the
+/// [`namada-masp-indexer`].
+///
+/// [`namada-masp-indexer`]: <https://github.com/anoma/namada-masp-indexer>
+#[cfg(not(target_family = "wasm"))]
+#[derive(Clone, Debug)]
+pub struct IndexerMaspClient {
+    indexer_api: Arc<reqwest::Url>,
+    client: reqwest::Client,
+}
+
+#[cfg(not(target_family = "wasm"))]
```

Comment:
Do we need this trait?

Reply:
well, we don't. it's more leaning towards an aesthetic change, rather than a functional one. it does kind of stop you from fucking up the keep-alive headers